app/controllers/orgs/rosters_controller.rb

```diff
@@ -146,6 +175,60 @@ def download_roster
     # rubocop:enable Metrics/MethodLength
     # rubocop:enable Metrics/AbcSize
 
+    def select_google_classroom
+      @google_classroom_courses = fetch_all_google_classrooms
+    end
+
+    # rubocop:disable Metrics/AbcSize
+    def search_google_classroom
+      courses_found = fetch_all_google_classrooms.select do |course|
+        course.name.downcase.include? params[:query].downcase
+      end
+
+      respond_to do |format|
+        format.html do
+          render partial: "orgs/rosters/google_classroom_collection",
+                 locals: { courses: courses_found }
+        end
+      end
+    end
+    # rubocop:enable Metrics/AbcSize
+
+    def import_from_google_classroom
+      students = list_google_classroom_students(params[:course_id])
+      return if students.nil?
+
+      if students.blank?
+        flash[:warning] = "No students were found in your Google Classroom. Please add students and try again."
+        redirect_to roster_path(current_organization)
+      else
+        current_organization.update_attributes!(google_course_id: params[:course_id])
+        add_google_classroom_students(students)
+      end
+    end
+
+    # rubocop:disable Metrics/AbcSize
+    def sync_google_classroom
+      return if current_organization.google_course_id.nil?
```

Comment:
That's a good question, not sure what to do here because if we're trying to sync a google classroom roster and we don't have an ID it's in a pretty broken state 🤔 I think if we just flash an error it should be fine?

Reply:
Flashing an error sounds like a good idea 👍